crates/tabby-common/src/config.rs

```diff
@@ -109,14 +113,12 @@ pub fn config_id_to_index(id: &str) -> Result<usize, anyhow::Error> {
 
 #[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
 pub struct RepositoryConfig {
-    git_url: String,
+    pub git_url: String,
+    #[serde(skip)]
+    pub source_id: String,
```

Comment:
```suggestion
```

Prefer not to put `source_id` in RepositoryConfig, RepositoryConfig should 1-1 map to its storage format in `configt.toml`.

You can still pass config_index_to_id when constructing `CodeRepository`

Reply:
OK